game-service/src/main/java/org/libertybikes/game/core/GameBoard.java

```diff
@@ -20,6 +24,7 @@
     public final Set<Obstacle> obstacles = new HashSet<>();
     public final Set<MovingObstacle> movingObstacles = new HashSet<>();
     public final Set<Player> players = new HashSet<>();
+    private final Map<Player, AI> aiMap = new HashMap<>();
```

Comment:
instead of GameBoard needing to maintain the mapping of Player->AI, it seems cleaner to add a `setAI(AI)` method on the `Player` class